engines/mohawk/riven.cpp

```diff
@@ -218,6 +215,11 @@ void MohawkEngine_Riven::doFrame() {
 	}
 
 	processInput();
+	
+	if (hasGameEnded()) {
```

Comment:
Will it launch on every frame?

Reply:
Yes, I like your proposed approach to put it right next to the event processor.

With the code here in the loop, I suspect that it might attempt to save several times at some condition.